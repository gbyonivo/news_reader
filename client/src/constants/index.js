/**
 * Enum for the different action stpyes
 * @enum {string}
 */
export const actionTypes = {
    LOAD_ARTICLES: 'LOAD_ARTICLES',
    CONNECTION_ERROR: 'CONNECTION_ERROR'
};
