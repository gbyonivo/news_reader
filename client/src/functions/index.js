export const filterList = (list, filterText, property) =>
  list.filter(item => item[property].toUpperCase().match(filterText.toUpperCase()));
