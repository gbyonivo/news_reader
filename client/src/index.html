<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" name="viewport" content="width=device-width, initial-scale=1"/>
    <title>Ivo</title>
</head>
<body>
    <div id="main-container">
    </div>
</body>
</html>