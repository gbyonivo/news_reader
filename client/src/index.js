import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import React from 'react';
import ReactDOM from 'react-dom';
import { Router, Route, browserHistory } from 'react-router';
import combinedReducers from './reducers/';
import Container from './container';

const store = applyMiddleware(thunk)(createStore)(combinedReducers);

const mainElement = document.getElementById('main-container');

ReactDOM.render(
    <Provider store={store}>
        <Router history={browserHistory}>
            <Route path="/" component={Container}/>
        </Router>
    </Provider>, mainElement
);
