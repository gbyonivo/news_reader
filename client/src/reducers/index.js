import { combineReducers } from 'redux';
import articles from './articles';
/**
 * @const
 * @type {object}
 */
const combinedReducers = combineReducers({
    articles
});

export default combinedReducers;
