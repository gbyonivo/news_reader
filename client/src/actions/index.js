import superagent from 'superagent';
import { actionTypes } from '../constants';

export const loadArticles = () => {
    return dispatch => {
        superagent
        .get('http://localhost:8000/articlesWithImages')
        .end((error, response) => {
            dispatch(error ?
            {
                type: actionTypes.CONNECTION_ERROR,
                error: 'Error loading articles'
            } : {
                type: actionTypes.LOAD_ARTICLES,
                articles: JSON.parse(response.text)
            });
        });
    };
};
