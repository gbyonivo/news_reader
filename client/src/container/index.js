import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as Actions from '../actions';
import Articles from '../components/articles';

class Container extends Component {
    componentDidMount() {
        this.props.actions.loadArticles();
    }
    render() {
        return (<div>
            <Articles/>
        </div>);
    }
}

Container.propTypes = {
    actions: PropTypes.object
};

const mapStateToProps = () => ({});


const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators(Actions, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Container);
