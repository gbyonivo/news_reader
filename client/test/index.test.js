import ReactDOM from 'react-dom';
import '../src';

jest.mock('react-dom', () => {
  return jest.genMockFromModule('react-dom');
});

describe('node', () => {
  it('matches snapshot', () => {
    const actual = ReactDOM.render.calls;
    expect(actual).toMatchSnapshot();
  });
});