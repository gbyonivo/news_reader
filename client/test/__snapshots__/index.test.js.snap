// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node matches snapshot 1`] = `undefined`;
